src/RequestExecutor/Response.ts

```diff
@@ -1,27 +1,34 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Protocol } from '../Handlers';
+
 export class Response {
-  public readonly status?: number;
+  public readonly protocol: Protocol;
+  public readonly status_code?: number;
   public readonly headers?: Record<string, string | string[]>;
   public readonly body?: string;
   public readonly message?: string;
-  public readonly errorCode?: string;
+  public readonly error_code?: string;
 
   constructor({
-    status,
+    protocol,
+    statusCode,
     headers,
     body,
     message,
     errorCode
   }: {
-    status?: number;
+    protocol: Protocol;
+    statusCode?: number;
     message?: string;
     errorCode?: string;
     headers?: Record<string, string | string[]>;
     body?: string;
   }) {
-    this.status = status;
+    this.protocol = protocol;
+    this.status_code = statusCode;
     this.headers = headers;
     this.body = body;
-    this.errorCode = errorCode;
+    this.error_code = errorCode;
```

Comment:
That's my fail. Let's use camelcase like before. We should use snake case only in `ForwardResponse`. BTW we have to extend `ForwardResponse` by new fields according to spec.